data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbSourcePartitionItem.java

```diff
@@ -23,13 +27,31 @@ public class DynamoDbSourcePartitionItem implements SourcePartitionStoreItem {
     private Instant partitionOwnershipTimeout;
     private Instant reOpenAt;
     private Long closedCount;
+    private String sourceStatusCombinationKey;
+    private String partitionPriority;
 
     @Override
     @DynamoDbPartitionKey
+    public String getSourceIdentifier() {
+        return sourceIdentifier;
+    }
+
+    @Override
+    @DynamoDbSortKey
     public String getSourcePartitionKey() {
         return sourcePartitionKey;
     }
 
+    @DynamoDbSecondaryPartitionKey(indexNames = "source-status")
+    public String getSourceStatusCombinationKey() {
+        return sourceStatusCombinationKey;
+    }
+
+    @DynamoDbSecondarySortKey(indexNames = "source-status")
```

Comment:
This looks wrong. copy-paste mistake? Also where is this `paritionPriority` being used? I see many `setPartitionPriority()` calls but don't see where it is being used.

Reply:
What looks wrong about this? It's correct. The partition priority as the secondary sort key will order partitions by this when the dynamodb query is run

Reply:
I thought same "source-status" being used in both cases and it's a copy-paste mistake. I guess not.